### DISCUSSION TITLE
Can I change the precision of weight only?

### DISCUSSION THREAD

**Discussion by returnwellbeing** [fastmachinelearning/hls4ml#1084] [Q&A] [answered]
Hi, I am thinking the idea of changing the precision of weights in linear layer. 
specifically I want to change only weight's precision and the input of linear layer is still in `model_default_t`
something is like that..
```
Input : ac_fixed<32, 16>
weight : ac_fixed<16, 8>
```
If there is any good idea, please let me know. thanks!

**Comment by vloncar** [comment, marked as answer]
Read the per-layer configuration section of the documentation.

**Reply by returnwellbeing** [reply]
Thank you for the reply. I found the solution from the documentation.
```py
config['LayerName']['linear']['Precision']['weight'] = 'ap_fixed<16,8>'
```
using configuration, I can change the precision of weight :)
